PhysicsTools/TensorFlow/src/NTSession.cc

```diff
@@ -701,6 +701,9 @@ Status NTSession::PRunSetup(const std::vector<string>& input_names,
   }
 
   *handle = run_state_args.handle;
+
+  delete barrier;
```

Comment:
from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/executor.h
```
// ExecutorBarrier deletes itself after the function returned by Get()
// is called.
```
it looks like the implementation does this deletion 
https://github.com/tensorflow/tensorflow/blob/98b9a4e45a559217bc89960b889c130af95c1d1a/tensorflow/core/common_runtime/executor.h#L216

I can imagine that this deletion may be skipped if somehow the loop over executors is empty.
Currently this change looks to me like double-delete. I'm not sure about the chances to have some run-time badness (since none was observed in the tests).

Reply:
That's a hard one... will take me some time to respond...

Reply:
it may be easier to leave it out of this PR and either open a separate PR or an issue.

Reply:
You are right. I will update this PR with that line removed later, in case there is not other comment in the evening. Thank you for your careful review by the way!